Code/RDGeneral/ConcurrentQueue.h

```diff
@@ -0,0 +1,132 @@
+//
+//  Copyright (C) 2020 Shrey Aryan
+//
+//   @@ All Rights Reserved @@
+//  This file is part of the RDKit.
+//  The contents are covered by the terms of the BSD license
+//  which is included in the file license.txt, found at the root
+//  of the RDKit source tree.
+//
+#ifdef RDK_THREADSAFE_SSS
+#ifndef CONCURRENT_QUEUE
+#define CONCURRENT_QUEUE
+#include <condition_variable>
+#include <thread>
+#include <vector>
+
+namespace RDKit {
+template <typename E>
+class ConcurrentQueue {
```

Comment:
Was there a reason not to use boost:lockfree:queue here?

Does the mutex guarantee same-ordering of output as input for instance?

Reply:
One cannot insert items that do not have a trivial assignment and destructor in the boost::lockfree::queue.

Reply:
Ah yes, good old std::string and it's non trivial constructor.

Reply:
@bp-kelley We've also seen a few posts indicating that the performance of `boost::lockfree::queue` isn't great.
In this case spending a lot of time waiting to add/remove things from the queue actually will make a difference.